include/llbuild/BuildSystem/BuildValue.h

```diff
@@ -52,15 +52,21 @@ class BuildValue {
 
     /// A value for a produced output whose command failed.
     FailedInput,
+      
+    /// A value for a produced output whose command was cancelled.
+    CancelledInput,
```

Comment:
You mean automatically compute this? I don't think so, but we could of course have better testing around it by capturing example encodings for each schema, and verifying we never get different encodings without also getting a new schema.